R/utility.R

```diff
@@ -98,57 +98,79 @@ getJobList <- function(jobIds = c()){
 #' }
 #' @export
 waitForNodesToComplete <- function(clusterId, timeout = 86400){
-  print("Booting compute nodes. . . ")
+  cat("Booting compute nodes. . . ", fill = TRUE)
 
-  pool <- getPool(clusterId)
+  pool <- rAzureBatch::getPool(clusterId)
+
+  # Validate the getPool request first, before setting the progress bar
+  if (!is.null(pool$code) && !is.null(pool$message)) {
+    stop(sprintf("Code: %s - Message: %s", pool$code, pool$message))
+  }
+
+  if (pool$targetDedicatedNodes + pool$targetLowPriorityNodes <= 0) {
+    stop("Pool count needs to be greater than 0.")
+  }
 
   numOfNodes <- pool$targetDedicatedNodes + pool$targetLowPriorityNodes
 
   pb <- txtProgressBar(min = 0, max = pool$targetDedicatedNodes + pool$targetLowPriorityNodes, style = 3)
   prevCount <- 0
   timeToTimeout <- Sys.time() + timeout
 
-  while(Sys.time() < timeToTimeout){
-    nodes <- listPoolNodes(clusterId)
+  while (Sys.time() < timeToTimeout) {
+    pool <- rAzureBatch::getPool(clusterId)
+
+    if (!is.null(pool$resizeErrors)) {
+      cat("\n")
+
+      resizeErrors <- ""
+      for (i in 1:length(pool$resizeErrors)) {
+        resizeErrors <- paste0(resizeErrors, sprintf("Code: %s - Message: %s \n", pool$resizeErrors[[i]]$code, pool$resizeErrors[[i]]$message))
+      }
+
+      stop(resizeErrors)
+    }
+
+    nodes <- rAzureBatch::listPoolNodes(clusterId)
 
     startTaskFailed <- TRUE
```

Comment:
It's cause we are using the stopifnot function, we can change it.. to just use the if statement. It's just an extra two lines of code